src/Razor/src/Microsoft.AspNetCore.Razor.Language/Legacy/CSharpCodeParser.cs

```diff
@@ -1756,14 +1756,12 @@ private void ParseExpectedCodeBlock(in SyntaxListBuilder<RazorSyntaxNode> builde
         {
             if (!EndOfFile)
             {
-                // Check for "{" to make sure we're at a block
-                if (!At(SyntaxKind.LeftBrace))
+                // See if we're at markup, if so, that's unsupported for single-line control flow statements.
+                if (!At(SyntaxKind.LeftBrace) && At(SyntaxKind.LessThan))
```

Comment:
Yup, that's all there is to the fix 😉

Reply:
is `<` the only indicator of Markup? What about cases like `@:` single line markup?

Reply:
Good catch on the single line markup. I shouldn't allow that.

Reply:
Also maybe template? `@<...`

Reply:
And `@@` double-transitions which is usually markup.

Reply:
`@<..>` is fine because it encompasses into a single statement. However `@@` needs to be handled.